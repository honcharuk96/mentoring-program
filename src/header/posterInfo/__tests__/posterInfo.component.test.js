import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PosterInfoComponent from '../posterInfo.component';

const shallow = new ShallowRenderer();
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: () => ({
    pathname: 'localhost:3000/example/path',
  }),
}));
const getPosterById = jest.fn();
const poster = {
  id: 337167,
  title: 'Fifty Shades Freed',
  tagline: "Don't miss the climax",
  vote_average: 6.1,
  vote_count: 1195,
  release_date: '2018-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  overview:
    'Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.',
  budget: 55000000,
  revenue: 136906000,
  genres: ['Drama', 'Romance'],
  runtime: 106,
};

describe('<PosterInfoComponent />', () => {
  it('should render PosterInfoComponent without poster', () => {
    const posterInfo = shallow.render(<PosterInfoComponent getPosterById={getPosterById} posterId={2} />);
    expect(posterInfo).toMatchSnapshot();
  });

  it('should render PosterInfoComponent with poster', () => {
    const posterInfo = shallow.render(
      <PosterInfoComponent getPosterById={getPosterById} posterId={2} poster={poster} />,
    );
    expect(posterInfo).toMatchSnapshot();
  });
});
