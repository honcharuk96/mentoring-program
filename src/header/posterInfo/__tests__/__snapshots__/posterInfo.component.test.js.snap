// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PosterInfoComponent /> should render PosterInfoComponent with poster 1`] = `
<posterInfostyled__PosterDetail>
  <posterInfostyled__PosterDetailRightBlock>
    <postersBlockstyled__Img
      alt="Fifty Shades Freed"
      loading="lazy"
      src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
    />
  </posterInfostyled__PosterDetailRightBlock>
  <posterInfostyled__PosterDetailLeftBlock>
    <posterInfostyled__PosterDetailHeader>
      <posterInfostyled__PosterDetailTitle>
        Fifty Shades Freed
      </posterInfostyled__PosterDetailTitle>
      <posterInfostyled__PosterDetailVoteAverage>
        6.1
      </posterInfostyled__PosterDetailVoteAverage>
    </posterInfostyled__PosterDetailHeader>
    <PosterCategoryComponent
      genres={
        Array [
          "Drama",
          "Romance",
        ]
      }
    />
    <posterInfostyled__PosterDetailTimes>
      <posterInfostyled__PosterDetailDate>
        2018
      </posterInfostyled__PosterDetailDate>
      <posterInfostyled__PosterDetailDate>
        1h 46min
      </posterInfostyled__PosterDetailDate>
    </posterInfostyled__PosterDetailTimes>
    <posterInfostyled__PosterDetailOverview>
      Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
    </posterInfostyled__PosterDetailOverview>
  </posterInfostyled__PosterDetailLeftBlock>
</posterInfostyled__PosterDetail>
`;

exports[`<PosterInfoComponent /> should render PosterInfoComponent without poster 1`] = `undefined`;
